src/protocols/LoRaWAN/LoRaWAN.cpp

```diff
@@ -2127,62 +2550,80 @@ size_t LoRaWANNode::execMacCommand(LoRaWANMacCommand_t* cmd) {
       this->rxDelays[0] = delay * 1000;
       this->rxDelays[1] = this->rxDelays[0] + 1000;
 
+#if !defined(RADIOLIB_EEPROM_UNSUPPORTED)
+    if(saveToEeprom) {
+      mod->hal->setPersistentParameter<uint8_t>(RADIOLIB_EEPROM_LORAWAN_RX_TIMING_SETUP_ID, cmd->payload[0]);
+      
+    }
+#endif
+
       // send the reply
       cmd->len = 0;
 
-      // TODO this should be sent repeatedly until the next downlink
-      pushMacCommand(cmd, &this->commandsUp);
-      
-      return(1);
+      // TODO send this repeatedly until a downlink is received
+      return(true);
     } break;
 
-    case(RADIOLIB_LORAWAN_MAC_CMD_TX_PARAM_SETUP): {
+    case(RADIOLIB_LORAWAN_MAC_TX_PARAM_SETUP): {
       uint8_t dlDwell = (cmd->payload[0] & 0x20) >> 5;
       uint8_t ulDwell = (cmd->payload[0] & 0x10) >> 4;
       uint8_t maxEirpRaw = cmd->payload[0] & 0x0F;
 
       // who the f came up with this ...
       const uint8_t eirpEncoding[] = { 8, 10, 12, 13, 14, 16, 18, 20, 21, 24, 26, 27, 29, 30, 33, 36 };
-      uint8_t maxEirp = eirpEncoding[maxEirpRaw];
-      RADIOLIB_DEBUG_PRINTLN("TX timing: dlDwell = %d, dlDwell = %d, maxEirp = %d dBm", dlDwell, ulDwell, maxEirp);
+      this->txPowerMax = eirpEncoding[maxEirpRaw];
+      RADIOLIB_DEBUG_PRINTLN("TX timing: dlDwell = %d, ulDwell = %d, maxEirp = %d dBm", dlDwell, ulDwell, this->txPowerMax);
 
-      // TODO implement this
-      (void)dlDwell;
-      (void)ulDwell;
-      (void)maxEirp;
-      return(1);
+      this->dwellTimeEnabledUp = ulDwell ? true : false;
+      this->dwellTimeUp = ulDwell ? 400 : 0;
+
+      this->dwellTimeEnabledDn = dlDwell ? true : false;
+      this->dwellTimeDn = dlDwell ? 400 : 0;
```

Comment:
If this is a LoraWAN-defined constant, it should be a macro since otherwise it's not clear where 400 came from.